ht.access

```diff
@@ -12,97 +12,348 @@ RewriteEngine On
+  AddType font/woff2                                  woff2
+  AddType application/vnd.ms-fontobject               eot
+  AddType font/ttf                                    ttf
+  AddType font/collection                             ttc
+  AddType font/otf                                    otf
+
+  # Other
+  AddType application/octet-stream                    safariextz
+  AddType application/x-bb-appworld                   bbaw
+  AddType application/x-chrome-extension              crx
+  AddType application/x-opera-extension               oex
+  AddType application/x-xpinstall                     xpi
+  AddType text/calendar                               ics
+  AddType text/markdown                               markdown md
+  AddType text/vcard                                  vcard vcf
+  AddType text/vnd.rim.location.xloc                  xloc
+  AddType text/vtt                                    vtt
+  AddType text/x-component                            htc
+
+</IfModule>
+
+
+# Character encodings
+
+# Serve all resources labeled as `text/html` or `text/plain`
+# with the media type `charset` parameter set to `UTF-8`.
+
+AddDefaultCharset utf-8
+
+
+# Serve the following file types with the media type `charset`
+# parameter set to `UTF-8`.
+
+<IfModule mod_mime.c>
+    AddCharset utf-8 .atom \
+                     .bbaw \
+                     .css \
+                     .geojson \
+                     .ics \
+                     .js \
+                     .json \
+                     .jsonld \
+                     .manifest \
+                     .markdown \
+                     .md \
+                     .mjs \
+                     .rdf \
+                     .rss \
+                     .topojson \
+                     .vtt \
+                     .webapp \
+                     .webmanifest \
+                     .xloc \
+                     .xml
+</IfModule>
+
 
 # Prevent dot directories (hidden directories like .git) to be exposed to the public
 # Except for the .well-known directory used by LetsEncrypt a.o
 RewriteRule "/\.|^\.(?!well-known/)" - [F]
 
+#Suppressing the `www.` at the beginning of URLs
+
+# Rewrite www.example.com -> example.com
 
-# Rewrite www.example.com -> example.com -- used with SEO Strict URLs plugin
-#RewriteCond %{HTTP_HOST} .
-#RewriteCond %{HTTP_HOST} ^www.(.*)$ [NC]
-#RewriteRule ^(.*)$ https://%1/$1 [R=301,L]
+# The same content should never be available under two different
+# URLs, especially not with and without `www.` at the beginning.
+# This can cause SEO problems (duplicate content), and therefore,
+# you should choose one of the alternatives and redirect the other
+# one.
 #
-# or for the opposite example.com -> www.example.com use the following
-# DO NOT USE BOTH
+# (!) NEVER USE BOTH WWW-RELATED RULES AT THE SAME TIME!
+
+# RewriteCond %{HTTPS} !=on
+# RewriteCond %{HTTP_HOST} ^www\.(.+)$ [NC]
+# RewriteRule ^ %{ENV:PROTO}://%1%{REQUEST_URI} [R=301,L]
```

Comment:
Can you use ENV:PROTO without first setting it manually? I've seen this used first before you can access that:
```
RewriteCond %{HTTPS} =on
RewriteRule ^(.*)$ - [env=proto:https]
RewriteCond %{HTTPS} !=on
RewriteRule ^(.*)$ - [env=proto:http]
```